htlcswitch/interceptable_switch.go

```diff
@@ -171,10 +195,25 @@ func (s *InterceptableSwitch) Stop() error {
 	close(s.quit)
 	s.wg.Wait()
 
+	s.blockEpochStream.Cancel()
+
 	return nil
 }
 
 func (s *InterceptableSwitch) run() {
+	// The block epoch stream will immediately stream the current height.
+	// Read it out here.
+	select {
+	case currentBlock := <-s.blockEpochStream.Epochs:
```

Comment:
`currentBlock, ok` and exit if `!ok` as below? I think we close this channel on shutdown, and setting our current height to 0 will be unexpected.